clang/lib/Driver/ToolChains/ClassicFlang.cpp

```diff
@@ -132,6 +132,11 @@ void ClassicFlang::ConstructJob(Compilation &C, const JobAction &JA,
     CommonCmdArgs.push_back("-nosave");
   }
 
+  for (auto Arg : Args.filtered(options::OPT_dM)) {
+    Arg->claim();
+    CommonCmdArgs.push_back("-list-macros");
```

Comment:
Do we need to pass `-list-macros` to `flang2`? Or just `flang1`?

Reply:
-list-macros is passed just to flang1.This is enough

Reply:
If `-list-macros` is only needed for `flang1`, then you should push it to `UpperCmdArgs` instead of `CommonCmdArgs`.

Reply:
@bryanpkc changed to `UpperCmdArgs`